roles/validate_inventory/tasks/vms.yml

```diff
@@ -0,0 +1,83 @@
+- name: Get all KVM Nodes
+  vars:
+    kvm_node_names: []
+  set_fact:
+    kvm_node_names: "{{ kvm_node_names + [item] }}"
+  when: hostvars[item]['vendor'] == 'KVM'
+  loop: "{{ groups['nodes'] }}"
+
+- name: Check there is no vm_host in services
+  assert:
+    that:
+      - ('vm_host' not in groups['services'])
+    fail_msg: >
+      The structure of the inventory has changed
+      please put vm_host into a group called vm_hosts
+      and add a vm_host entry to the KVM node to point
+      to the host of the vm.
+  when: (kvm_node_names is defined) and (kvm_node_names | length > 0)
```

Comment:
nice idea using the validation to inform of a change to the Inventory. I don't think this needs to be conditional, though. It's a sensible check even if no VMs are defined.